integration-tests/flyway/src/main/java/io/quarkus/example/flyway/FlywayFunctionalityResource.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2018 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.quarkus.example.flyway;
+
+import static java.sql.DriverManager.getConnection;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.flywaydb.core.Flyway;
+import org.jboss.logging.Logger;
+
+@Path("/flyway")
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+public class FlywayFunctionalityResource {
+    private static final Logger LOGGER = Logger.getLogger(FlywayFunctionalityResource.class);
+    @ConfigProperty(name = "datasource.url")
+    String dbURL;
+    @ConfigProperty(name = "datasource.username")
+    String dbUser;
+    @ConfigProperty(name = "datasource.password")
+    String dbPassword;
+
+    @GET
+    @Path("/migrate")
+    public String doMigrate() throws SQLException {
+        Flyway flyway = Flyway.configure()
+                .dataSource(dbURL, dbUser, dbPassword)
+                .load();
+        InputStream resource = this.getClass().getClassLoader().getResourceAsStream(
```

Comment:
I was wondering if reading the migration file is needed here. Knowing that is is only used for logging purposes. Maybe we can delete this.

Reply:
Yes, you are right, it was part of my early tests to understand the native resource access vs the JVM classloader access. I will remove it.